### PR TITLE
Adding TC for SharedBackupDelete

### PR DESCRIPTION
**Pull request by skonda-px**
**What this PR does / why we need it**:
Adding TC for Share and delete from all backups page

**Which issue(s) this PR fixes** (optional)
Closes #PA-511

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/backup/backup_test.go [hunk 1]
```diff
@@ -1129,6 +1129,214 @@ var _ = Describe("{ShareBackupAndEdit}", func() {
 	})
 })
 
+func createUsers(numberOfUsers int) []string {
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:1132]: Move this function to the bottom of the file

File: tests/backup/backup_test.go [hunk 2]
```diff
@@ -1129,6 +1129,214 @@ var _ = Describe("{ShareBackupAndEdit}", func() {
...
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+	It("Share the backups and delete", func() {
+		Step("Validate applications and get their labels", func() {
+			log.InfoD("Validate applications and get their labels")
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:1191]: We are just validating the applications, we are not getting the labels.

File: tests/backup/backup_test.go [hunk 3]
```diff
@@ -1129,6 +1129,214 @@ var _ = Describe("{ShareBackupAndEdit}", func() {
...
+							nil, orgID, clusterUid, "", "", "", "", ctx)
+					}(backupName)
+				}
+				wg.Wait()
+			}
+			log.Infof("List of backups - %v", backupNames)
+		})
+		Step("Share backup with multiple users", func() {
+			log.InfoD("Share backup with multiple users")
+			log.Infof("Sharing backup with  multiple users")
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:1250]: log.InfoD is sufficient here. Remove the log.Infof

File: tests/backup/backup_test.go [hunk 4]
```diff
@@ -1129,6 +1129,214 @@ var _ = Describe("{ShareBackupAndEdit}", func() {
...
+				for _, backup := range backupNames {
+					// Get Backup UID
+					backupDriver := Inst().Backup
+					backupUID, err := backupDriver.GetBackupUID(ctx, backup, orgID)
+					log.FailOnError(err, "Failed while trying to get backup UID for - %s", backup)
+
+					// Delete backup to confirm that the user cannot delete the backup
+					_, err = DeleteBackup(backup, backupUID, orgID, ctxNonAdmin)
+					log.Infof("Error message - %s", err.Error())
+					dash.VerifyFatal(strings.Contains(err.Error(), "doesn't have permission to delete backup"), true, "Verifying backup deletion is not possible")
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:1277]: Also need to verify that the users cannot restore the backup with view only access

File: tests/backup/backup_test.go [hunk 5]
```diff
@@ -1129,6 +1129,214 @@ var _ = Describe("{ShareBackupAndEdit}", func() {
...
+					backupUID, err := backupDriver.GetBackupUID(ctx, backup, orgID)
+					log.FailOnError(err, "Failed while trying to get backup UID for - %s", backup)
+					_, err = DeleteBackup(backup, backupUID, orgID, ctx)
+					log.FailOnError(err, "Failed to delete backup - %s", backup)
+					err = backupDriver.WaitForBackupDeletion(ctx, backup, orgID, time.Minute*10, time.Minute*1)
+					log.FailOnError(err, "Error waiting for backup deletion %v", backup)
+				}(backup)
+			}
+			wg.Wait()
+		})
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:1299]: Missed the step where we need to validate that the users who had view only access cannot enumerate the backups after deletion

**Comment by mkoppal-px** [comment]
I do not see the aetos dashboard link

**Comment by skonda-px** [comment]
http://aetos.pwx.purestorage.com/resultSet/testSetID/78678

File: tests/backup/backup_test.go (hunk 5 above)
Comment (skonda-px) [reply tests/backup/backup_test.go:1838]: Done

File: tests/backup/backup_test.go (hunk 4 above)
Comment (skonda-px) [reply tests/backup/backup_test.go:1802]: Done

File: tests/backup/backup_test.go (hunk 3 above)
Comment (skonda-px) [reply tests/backup/backup_test.go]: Removed

File: tests/backup/backup_test.go (hunk 1 above)
Comment (skonda-px) [reply tests/backup/backup_test.go]: Moved to bottom of the file

File: tests/backup/backup_test.go [hunk 6]
```diff
@@ -1666,6 +1666,222 @@ var _ = Describe("{ShareBackupAndEdit}", func() {
...
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+	It("Share the backups and delete", func() {
+		Step("Validate applications", func() {
+			log.InfoD("Validate applications")
+			ValidateApplications(contexts)
+			log.Infof("Create list of pod selector for the apps deployed")
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:1709]: Remove this line

File: tests/backup/backup_test.go [hunk 7]
```diff
@@ -1666,6 +1666,222 @@ var _ = Describe("{ShareBackupAndEdit}", func() {
...
+					restoreName := fmt.Sprintf("%s-%v", RestoreNamePrefix, time.Now().Unix())
+					err = CreateRestore(restoreName, backup, nil, destinationClusterName, orgID, ctxNonAdmin)
+
+					// Restore validation to make sure that the user with cannot restore
+					dash.VerifyFatal(strings.Contains(err.Error(), "failed to retrieve backup location"), true, "Verifying backup restore is not possible")
+
+					// Delete backup to confirm that the user cannot delete the backup
+					_, err = DeleteBackup(backup, backupUID, orgID, ctxNonAdmin)
+					log.Infof("Error message - %s", err.Error())
+					dash.VerifyFatal(strings.Contains(err.Error(), "doesn't have permission to delete backup"), true, "Verifying backup deletion is not possible")
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:1802]: Add the backup name here

File: tests/backup/backup_test.go [hunk 8]
```diff
@@ -1666,6 +1666,222 @@ var _ = Describe("{ShareBackupAndEdit}", func() {
...
+					backupDriver := Inst().Backup
+					backupUID, err := backupDriver.GetBackupUID(ctx, backup, orgID)
+					log.FailOnError(err, "Failed while trying to get backup UID for - %s", backup)
+
+					// Start Restore
+					restoreName := fmt.Sprintf("%s-%v", RestoreNamePrefix, time.Now().Unix())
+					err = CreateRestore(restoreName, backup, nil, destinationClusterName, orgID, ctxNonAdmin)
+
+					// Restore validation to make sure that the user with cannot restore
+					dash.VerifyFatal(strings.Contains(err.Error(), "failed to retrieve backup location"), true, "Verifying backup restore is not possible")
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:1797]: Add the backup and restore name here

File: tests/backup/backup_test.go (hunk 6 above)
Comment (skonda-px) [reply tests/backup/backup_test.go:1709]: Removed

File: tests/backup/backup_test.go (hunk 8 above)
Comment (skonda-px) [reply tests/backup/backup_test.go:1797]: Added backup and restore name.

**Review by kshithijiyer-px — APPROVED** [review]

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**